app/controllers/hosts_controller.rb

```diff
@@ -448,7 +448,7 @@ def submit_rebuild_config
   end
 
   def submit_multiple_build
-    @hosts.delete_if do |host|
+    @hosts.to_a.delete_if do |host|
```

Comment:
Are the rest of the actions in this controller safe?  We have little to no test coverage of them, so unless they've actually been tried, might they need work too?